### ISSUE TITLE
Unclear base directory of relative paths to external dictionary files

### ISSUE THREAD

**Issue by zertrin** [valentjn/vscode-ltex#146]
**Describe the bug**

In the current version, it seems that the words defined in an external dictionary file are not used for ignoring spell-checking errors.

It is not shown in the logs below, but I made sure that the external dictionary is actually found by `vscode-ltex` by setting the path to a wrong file name (`spell-check_dict.txt` which does not exist) and verifying in the log that there is a file not found exception, and then by setting the value back to the correct file name (`spellcheck_dict.txt`) and noticing that there is no more file not found error.

**Steps to reproduce**

Minimal reproducible example:

`document.tex`

```tex
\documentclass{article}
\usepackage[utf8]{inputenc}

\begin{document}

This is a minimal working example to demonstrate a bug in vscode-ltex.

This word does not exist: nqazwsx.
But it is in the external dictionary file so it should be ignored.

\end{document}
```

`spellcheck_dict.txt`

```plaintext
nqazwsx
```

`.vscode/settings.json`

```json
{
    "ltex.workspaceFolderDictionary": {
        "en-US": [
            ":spellcheck_dict.txt"
        ],
    },
    "ltex.ltex-ls.logLevel": "finest"
}
```

**Expected behavior**

In the minimal example above, only `vscode-ltex` should be detected as a problem. But `nqazwsx` which is in the external spellcheck dictionary is incorrectly indicated as a problem...

Words defined in an external dictionary should be ignored during spell checking.

**LTeX configuration**

<details>

```
{
  "enabled": [
    "markdown",
    "latex",
    "rsweave"
  ],
  "language": "en-US",
  "dictionary": {},
  "workspaceDictionary": {},
  "workspaceFolderDictionary": {
    "en-US": [
      ":spellcheck_dict.txt"
    ]
  },
  "disabledRules": {},
  "workspaceDisabledRules": {},
  "workspaceFolderDisabledRules": {},
  "enabledRules": {},
  "workspaceEnabledRules": {},
  "workspaceFolderEnabledRules": {},
  "ltex-ls": {
    "path": "",
    "languageToolHttpServerUri": "",
    "logLevel": "finest"
  },
  "java": {
    "path": "",
    "initialHeapSize": 64,
    "maximumHeapSize": 512
  },
  "commands": {
    "ignore": [],
    "dummy": []
  },
  "environments": {
    "ignore": []
  },
  "markdown": {
    "ignore": [
      "CodeBlock",
      "FencedCodeBlock",
      "IndentedCodeBlock"
    ],
    "dummy": [
      "AutoLink",
      "Code"
    ]
  },
  "ignoreRuleInSentence": [],
  "configurationTarget": {
    "addToDictionary": "workspaceFolder",
    "disableRule": "workspaceFolder",
    "ignoreRuleInSentence": "workspaceFolder"
  },
  "additionalRules": {
    "motherTongue": "",
    "languageModel": "",
    "neuralNetworkModel": "",
    "word2VecModel": ""
  },
  "sentenceCacheSize": 2000,
  "diagnosticSeverity": "information",
  "clearDiagnosticsWhenClosingFile": true,
  "trace": {
    "server": "off"
  }
}
```

</details>

**"LTeX Language Server" log file**

<details>

```
[Info  - 12:33:32 PM] Starting ltex-ls...
Oct 23, 2020 12:33:37 PM org.bsplines.ltexls.server.LtexLanguageServer initialize
INFO: ltex-ls 7.3.1 - initializing...
Oct 23, 2020 12:33:37 PM org.bsplines.ltexls.server.LtexLanguageServer initialize
INFO: Setting locale to 'en'
Oct 23, 2020 12:33:37 PM org.bsplines.ltexls.settings.SettingsManager logDifferentSettings
FINE: Reinitializing LanguageTool due to different settings for language 'en-US': setting 'commands.dummy', old 'null', new '[]'; setting 'commands.ignore', old 'null', new '[]'; setting 'additionalRules.neuralNetworkModel', old 'null', new ''; setting 'language', old 'null', new 'en-US'; setting 'clearDiagnosticsWhenClosingFile', old 'null', new 'true'; setting 'markdown.ignore', old 'null', new '[IndentedCodeBlock, CodeBlock, FencedCodeBlock]'; setting 'markdown.dummy', old 'null', new '[Code, AutoLink]'; setting 'dictionary', old 'null', new '{en-US=[:spellcheck_dict.txt]}'; setting 'disabledRules', old 'null', new '{}'; setting 'ltex-ls.languageToolHttpServerUri', old 'null', new ''; setting 'diagnosticSeverity', old 'null', new 'Information'; setting 'fullDictionary', old '[]', new '[:spellcheck_dict.txt]'; setting 'sentenceCacheSize', old 'null', new '2000'; setting 'environments.ignore', old 'null', new '[]'; setting 'ignoreRuleInSentence', old 'null', new '[]'; setting 'additionalRules.languageModel', old 'null', new ''; setting 'additionalRules.motherTongue', old 'null', new ''; setting 'ltex-ls.logLevel', old 'null', new 'FINEST'; setting 'additionalRules.word2VecModel', old 'null', new ''; setting 'enabled', old 'null', new '[rsweave, markdown, latex]'; setting 'enabledRules', old 'null', new '{}'
Oct 23, 2020 12:33:38 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINER: Checking the following text in language 'en-US' via LanguageTool: " \n\n\n\nThis is a minimal working example to demonstrate a bug in vscode-ltex.\n\nThis word does not exist: nqazwsx. But it is in the external dictionary file so it should be ignored.\n\n"
Oct 23, 2020 12:33:38 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINEST: annotatedTextParts = [MARKUP("\\documentclass{article}"), MARKUP("\n"), FAKE_CONTENT(" "), MARKUP("\\usepackage[utf8]{inputenc}"), MARKUP("\n\n"), FAKE_CONTENT("\n\n"), MARKUP("\\begin"), MARKUP("{document}"), MARKUP("\n\n"), FAKE_CONTENT("\n\n"), TEXT("T"), TEXT("h"), TEXT("i"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("a"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("m"), TEXT("i"), TEXT("n"), TEXT("i"), TEXT("m"), TEXT("a"), TEXT("l"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("w"), TEXT("o"), TEXT("r"), TEXT("k"), TEXT("i"), TEXT("n"), TEXT("g"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("e"), TEXT("x"), TEXT("a"), TEXT("m"), TEXT("p"), TEXT("l"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("t"), TEXT("o"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("d"), TEXT("e"), TEXT("m"), TEXT("o"), TEXT("n"), TEXT("s"), TEXT("t"), TEXT("r"), TEXT("a"), TEXT("t"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("a"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("b"), TEXT("u"), TEXT("g"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("n"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("v"), TEXT("s"), TEXT("c"), TEXT("o"), TEXT("d"), TEXT("e"), TEXT("-"), TEXT("l"), TEXT("t"), TEXT("e"), TEXT("x"), TEXT("."), MARKUP("\n\n"), FAKE_CONTENT("\n\n"), TEXT("T"), TEXT("h"), TEXT("i"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("w"), TEXT("o"), TEXT("r"), TEXT("d"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("d"), TEXT("o"), TEXT("e"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("n"), TEXT("o"), TEXT("t"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("e"), TEXT("x"), TEXT("i"), TEXT("s"), TEXT("t"), TEXT(":"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("n"), TEXT("q"), TEXT("a"), TEXT("z"), TEXT("w"), TEXT("s"), TEXT("x"), TEXT("."), MARKUP("\n"), FAKE_CONTENT(" "), TEXT("B"), TEXT("u"), TEXT("t"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("t"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("n"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("t"), TEXT("h"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("e"), TEXT("x"), TEXT("t"), TEXT("e"), TEXT("r"), TEXT("n"), TEXT("a"), TEXT("l"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("d"), TEXT("i"), TEXT("c"), TEXT("t"), TEXT("i"), TEXT("o"), TEXT("n"), TEXT("a"), TEXT("r"), TEXT("y"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("f"), TEXT("i"), TEXT("l"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("s"), TEXT("o"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("t"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("s"), TEXT("h"), TEXT("o"), TEXT("u"), TEXT("l"), TEXT("d"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("b"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("g"), TEXT("n"), TEXT("o"), TEXT("r"), TEXT("e"), TEXT("d"), TEXT("."), MARKUP("\n\n"), FAKE_CONTENT("\n\n"), MARKUP("\\end"), MARKUP("{document}")]
Oct 23, 2020 12:33:38 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINER: matchesCache.size() = 0
Oct 23, 2020 12:33:38 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINER: remoteMatchesCache.size() = 0
Oct 23, 2020 12:33:38 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINER: sentenceCache.size() = 0
Oct 23, 2020 12:33:38 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINEST: matchesCache = {}
Oct 23, 2020 12:33:38 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINEST: remoteMatchesCache = {}
Oct 23, 2020 12:33:38 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINEST: sentenceCache = {}
Oct 23, 2020 12:33:44 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINER: Checking done in 6,013ms
Oct 23, 2020 12:33:44 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINE: Obtained 2 rule matches
```

</details>

**"LTeX Language Client" log file**

<details>

```
2020-10-23T04:33:31.541Z Info: Setting LTeX UI language to 'en'.
2020-10-23T04:33:31.542Z Info: Loading i18n messages...
2020-10-23T04:33:31.549Z Info: Loading default i18n messages...
2020-10-23T04:33:31.552Z Info: 
2020-10-23T04:33:31.553Z Info: ltex.ltex-ls.path not set.
2020-10-23T04:33:31.553Z Info: Searching for ltex-ls in '/home/redacted/.vscode-server/extensions/valentjn.vscode-ltex-7.3.1/lib'...
2020-10-23T04:33:31.554Z Info: ltex-ls found in '/home/redacted/.vscode-server/extensions/valentjn.vscode-ltex-7.3.1/lib/ltex-ls-7.3.1'.
2020-10-23T04:33:31.554Z Info: 
2020-10-23T04:33:31.554Z Info: ltex.java.path not set.
2020-10-23T04:33:31.554Z Info: Using ltex-ls from '/home/redacted/.vscode-server/extensions/valentjn.vscode-ltex-7.3.1/lib/ltex-ls-7.3.1'.
2020-10-23T04:33:31.554Z Info: Using Java from PATH or JAVA_HOME (may fail if not installed).
2020-10-23T04:33:31.694Z Info: Testing ltex-ls...
2020-10-23T04:33:31.694Z Info:   Command: "/home/redacted/.vscode-server/extensions/valentjn.vscode-ltex-7.3.1/lib/ltex-ls-7.3.1/bin/ltex-ls"
2020-10-23T04:33:31.694Z Info:   Arguments: ["--version"]
2020-10-23T04:33:31.694Z Info:   env['JAVA_HOME']: undefined
2020-10-23T04:33:31.694Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-10-23T04:33:32.852Z Info: Test successful!
2020-10-23T04:33:32.893Z Info: 
2020-10-23T04:33:32.907Z Info: Starting ltex-ls...
2020-10-23T04:33:32.907Z Info:   Command: "/home/redacted/.vscode-server/extensions/valentjn.vscode-ltex-7.3.1/lib/ltex-ls-7.3.1/bin/ltex-ls"
2020-10-23T04:33:32.907Z Info:   Arguments: []
2020-10-23T04:33:32.907Z Info:   env['JAVA_HOME']: undefined
2020-10-23T04:33:32.907Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-10-23T04:33:32.907Z Info: 
2020-10-23T04:34:00.160Z Info: Creating bug report...
```

</details>

**Version information**
List here the version information of the relevant software.

- Operating system: Linux (linux), x64, 4.4.0-18362-Microsoft
- VS Code: 1.50.1
- vscode-ltex: 7.3.1
- ltex-ls: 7.3.1
- Java: 11.0.8

**Comment by valentjn** [comment]
I think relative paths are from the extension folder. Try an absolute one instead.

**Comment by zertrin** [comment]
But that does not make sense with the fact that purposefully setting a wrong file name results in a file not found exception, whereas setting it as it is there is no exception, **strongly indicating that in the case above the file was correctly found**.

See case where I purposefully give a wrong name (`":spellcheck_dict_notexistent.txt"` in the settings file):

```plaintext
Oct 23, 2020 1:00:30 PM org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint notify
INFO: Unsupported notification method: $/setTraceNotification
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.tools.Tools readFile
WARNING: Could not read file '/home/redacted/tmp/vscode-ltex-bug-ext-dict/spellcheck_dict_notexistent.txt'. The following exception occurred:
java.nio.file.NoSuchFileException: /home/redacted/tmp/vscode-ltex-bug-ext-dict/spellcheck_dict_notexistent.txt
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:370)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:421)
	at java.base/java.nio.file.Files.readAllBytes(Files.java:3205)
	at org.bsplines.ltexls.tools.Tools.readFile(Tools.java:162)
	at org.bsplines.ltexls.settings.DictionaryFileWatcher.setWatchedDictionary(DictionaryFileWatcher.java:106)
	at org.bsplines.ltexls.settings.SettingsManager.setDictionaryFileWatcher(SettingsManager.java:163)
	at org.bsplines.ltexls.settings.SettingsManager.setSettings(SettingsManager.java:132)
	at org.bsplines.ltexls.settings.SettingsManager.setSettings(SettingsManager.java:117)
	at org.bsplines.ltexls.server.LtexTextDocumentItem.lambda$check$2(LtexTextDocumentItem.java:412)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:212)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.settings.SettingsManager logDifferentSettings
FINE: Reinitializing LanguageTool due to different settings for language 'en-US': setting 'dictionary', old '{en-US=[:spellcheck_dict.txt]}', new '{en-US=[:spellcheck_dict_notexistent.txt]}'; setting 'fullDictionary', old '[:spellcheck_dict.txt]', new '[:spellcheck_dict_notexistent.txt]'
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINER: Checking the following text in language 'en-US' via LanguageTool: " \n\n\n\nThis is a minimal working example to demonstrate a bug in vscode-ltex.\n\nThis word does not exist: nqazwsx. But it is in the external dictionary file so it should be ignored.\n\n"
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINEST: annotatedTextParts = [MARKUP("\\documentclass{article}"), MARKUP("\n"), FAKE_CONTENT(" "), MARKUP("\\usepackage[utf8]{inputenc}"), MARKUP("\n\n"), FAKE_CONTENT("\n\n"), MARKUP("\\begin"), MARKUP("{document}"), MARKUP("\n\n"), FAKE_CONTENT("\n\n"), TEXT("T"), TEXT("h"), TEXT("i"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("a"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("m"), TEXT("i"), TEXT("n"), TEXT("i"), TEXT("m"), TEXT("a"), TEXT("l"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("w"), TEXT("o"), TEXT("r"), TEXT("k"), TEXT("i"), TEXT("n"), TEXT("g"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("e"), TEXT("x"), TEXT("a"), TEXT("m"), TEXT("p"), TEXT("l"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("t"), TEXT("o"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("d"), TEXT("e"), TEXT("m"), TEXT("o"), TEXT("n"), TEXT("s"), TEXT("t"), TEXT("r"), TEXT("a"), TEXT("t"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("a"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("b"), TEXT("u"), TEXT("g"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("n"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("v"), TEXT("s"), TEXT("c"), TEXT("o"), TEXT("d"), TEXT("e"), TEXT("-"), TEXT("l"), TEXT("t"), TEXT("e"), TEXT("x"), TEXT("."), MARKUP("\n\n"), FAKE_CONTENT("\n\n"), TEXT("T"), TEXT("h"), TEXT("i"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("w"), TEXT("o"), TEXT("r"), TEXT("d"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("d"), TEXT("o"), TEXT("e"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("n"), TEXT("o"), TEXT("t"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("e"), TEXT("x"), TEXT("i"), TEXT("s"), TEXT("t"), TEXT(":"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("n"), TEXT("q"), TEXT("a"), TEXT("z"), TEXT("w"), TEXT("s"), TEXT("x"), TEXT("."), MARKUP("\n"), FAKE_CONTENT(" "), TEXT("B"), TEXT("u"), TEXT("t"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("t"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("s"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("n"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("t"), TEXT("h"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("e"), TEXT("x"), TEXT("t"), TEXT("e"), TEXT("r"), TEXT("n"), TEXT("a"), TEXT("l"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("d"), TEXT("i"), TEXT("c"), TEXT("t"), TEXT("i"), TEXT("o"), TEXT("n"), TEXT("a"), TEXT("r"), TEXT("y"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("f"), TEXT("i"), TEXT("l"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("s"), TEXT("o"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("t"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("s"), TEXT("h"), TEXT("o"), TEXT("u"), TEXT("l"), TEXT("d"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("b"), TEXT("e"), MARKUP(" "), FAKE_CONTENT(" "), TEXT("i"), TEXT("g"), TEXT("n"), TEXT("o"), TEXT("r"), TEXT("e"), TEXT("d"), TEXT("."), MARKUP("\n\n"), FAKE_CONTENT("\n\n"), MARKUP("\\end"), MARKUP("{document}")]
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINER: matchesCache.size() = 0
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINER: remoteMatchesCache.size() = 0
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINER: sentenceCache.size() = 0
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINEST: matchesCache = {}
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINEST: remoteMatchesCache = {}
Oct 23, 2020 1:00:30 PM org.bsplines.ltexls.languagetool.LanguageToolJavaInterface check
FINEST: sentenceCache = {}
Oct 23, 2020 1:00:33 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINER: Checking done in 2,223ms
Oct 23, 2020 1:00:33 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINE: Obtained 2 rule matches
```

EDIT: note that the search path was correct, the document to check as well as the dictionary are both located in `/home/redacted/tmp/vscode-ltex-bug-ext-dict/`.

**Comment by zertrin** [comment]
Ok I used an absolute path and it worked this time.

But this is very weird, there is no indication that the relative path was wrong... So the file was somehow found, but the content ignored?

I would still consider this as some form of bug:
- either relative paths are not supported and there should be an error message and some explanation for the user
- or relative path handling should be fixed to work properly

Thanks for the quick reply ❤️!

**Comment by valentjn** [comment]
It's neither. Relative paths are supported, of course. At the end of the day, the path is probably fed somewhere to some syscall, which will resolve the relative path, and there's no logic in LT<sub>E</sub>X that detects relative paths and prevents their usage before that. It's just not clear currently (even I'm not 100% sure) what the base directory is. So yes, this is a bug, but in the documentation, which should state how relative paths are resolved.

This should be fixed now. Depending on the scope of the settings (user, workspace, workspace folder) in which you include the external file, relative paths are relative to the `.vscode` directory of the corresponding `settings.json` (except for the user scope, then they're relative to the global storage path of the extension).

The documentation will be updated to reflect this.

**Comment by valentjn** [comment]
Fix released in 8.0.0, see [external setting files](https://valentjn.github.io/vscode-ltex/docs/advanced-usage.html#external-setting-files) in the docs.

**Comment by zertrin** [comment]
I have upgraded to 8.0.0 today, and am now able to use relative paths (defined in `.vscode/settings.json`, now correctly relative to `.vscode`).

Example, now working:

```json
    "ltex.dictionary": {
        "en-GB": [
            ":../spellcheck_dict/en-GB.txt",
            ":../spellcheck_dict/abbr_acr.txt",
            ":../spellcheck_dict/proper_names.txt"
        ],
        "de-DE": [
            ":../spellcheck_dict/de-DE.txt",
            ":../spellcheck_dict/abbr_acr.txt",
            ":../spellcheck_dict/proper_names.txt"
        ]
    },
```

Thanks!
